src/synthesizer/typescript_synthesizer.rs

```diff
@@ -12,155 +12,193 @@ pub struct TypescriptSynthesizer {
+            ));
 
-            for (name, prop) in reference.properties.iter() {
+            for (i, (name, prop)) in reference.properties.iter().enumerate() {
                 match to_string_ir(prop) {
                     None => {}
                     Some(x) => {
-                        println!("\t{}:{},", camel_case(name), x);
+                        append_str_with_linebreak(output, &format!(
+                            "{}: {}{}",
+                            camel_case(name),
+                            x,
+                            // Remove trailing comma.
+                            if i == reference.properties.len() - 1 { "" } else { "," }
+                        ));
                     }
                 }
             }
-            println!("}});");
+
+            append_str_with_linebreak(output, "\t\t});");
 
             if let Some(metadata) = &reference.metadata {
-                println!("{}.addOverride('Metadata', ", camel_case(&reference.name));
+                append_str_with_linebreak(output, &format!("{}.addOverride('Metadata', ", camel_case(&reference.name)));
+
                 match to_string_ir(metadata) {
                     None => panic!("This should never fail"),
                     Some(x) => {
-                        println!("{}", x);
+                        append_str_with_linebreak(output, &format!("{}", x));
                     }
                 };
 
-                println!(");");
+                append_str_with_linebreak(output, ");");
             }
+
             if let Some(update_policy) = &reference.update_policy {
-                println!(
+                append_str_with_linebreak(output, &format!(
                     "{}.addOverride('UpdatePolicy', ",
-                    camel_case(&reference.name)
-                );
+                    camel_case(&reference.name),
+                ));
+
                 match to_string_ir(update_policy) {
                     None => panic!("This should never fail"),
                     Some(x) => {
-                        println!("{}", x);
+                        append_str_with_linebreak(output, &format!("{}", x));
                     }
                 };
-                println!(");");
+
+                append_str_with_linebreak(output, ");");
             }
 
             if let Some(deletion_policy) = &reference.deletion_policy {
-                println!(
+                append_str_with_linebreak(output, &format!(
                     "{}.addOverride('DeletionPolicy', '{}');",
                     camel_case(&reference.name),
-                    deletion_policy
-                );
+                    deletion_policy,
+                ));
             }
+
             if !reference.dependencies.is_empty() {
-                println!("{}.addOverride('DependsOn', [", camel_case(&reference.name));
+                append_str_with_linebreak(output, &format!("{}.addOverride('DependsOn', [", camel_case(&reference.name)));
```

Comment:
For brevity, would `line_break` be better? Since the first one is `output` gives away the goal.